Import latest python scripts and MCUBoot image

urutva opened the pull request:
<!--
For more information on the requirements for pull requests, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html).

NOTE: Do not remove any of the template headings (even for optional sections) as this
template is automatically parsed. 
-->

### Summary of changes <!-- Required -->
To help with the integration of Musca B1 into Mbed OS, python
signing scripts and MCUBoot image and RSA private key for Musca A
has been updated from latest TF-M
(https://git.trustedfirmware.org/trusted-firmware-m.git/commit/?id=6c5be4a98e4d7055ee49076ca4e515fb4b172e66).

Signed-off-by: Devaraj Ranganna <devaraj.ranganna@arm.com>
<!-- 
    Please provide the following information: 

    Description of the the change (what is this fixing / adding / removing?).

    Why the change is needed (if this is fixing a reported issue please summarize what
    the issue is and add the reference. E.g. Fixes #17119).

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
    
-->

#### Impact of changes <!-- Optional -->
<!-- 
    If there are any implications for users taking this change then they must be 
    provided here. For Major PR types this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

#### Migration actions required <!-- Optional -->
<!-- 
    This should only be applicable in Major PR types for which this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

### Documentation <!-- Required -->

<!-- 
    Please provide details of any document updates required, including links to any
    related PRs against the docs repository.
    If no document updates are required please specify 'None', this at least tells us
    that this has been considered.
-->

----------------------------------------------------------------------------------------------------------------
### Pull request type <!-- Required -->

<!--
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front of them would change the meaning incorrectly. 
-->
    [x] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results <!-- Required -->

<!--
    Provide all the information required, listing all the testing performed. For new targets please attach full test results for all supported compilers.
-->
    [] No Tests required for this change (E.g docs only update)
    [x] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers <!-- Optional -->
@Patater @jainvikas8 
<!--
    Request additional reviewers with @username or @team
-->

----------------------------------------------------------------------------------------------------------------

0xc0170 commented on targets/targets.json:
```diff
@@ -5757,7 +5757,7 @@
         "default_toolchain": "ARMC6",
         "extra_labels": ["ARM_SSG", "MUSCA_A1"],
-        "forced_reset_timeout": 7,
+        "forced_reset_timeout": 13,
```
this should be a separate commit, or even a pull request?

urutva replied:
@0xc0170 This workaround was done to fix the bug in TF-M bootloader. But it is already fixed in the version of TF-M bootloader imported into Mbed OS. Therefore we don't need this workaround anymore.

0xc0170 commented on tools/psa/tfm/bin_utils/imgtool_lib/boot_record.py:
```diff
@@ -0,0 +1,77 @@
+
+# Copyright (c) 2019, Arm Limited.
+#
```
add SPDX identifier

urutva replied:
@0xc0170 Fixed

0xc0170 reviewed (dismissed):
Please also rebase to resolve a conflict

urutva committed:
Import latest python scripts and MCUBoot image

To help with the integration of Musca B1 into Mbed OS, python
signing scripts and MCUBoot image and RSA private key for Musca A
has been updated from latest TF-M
(https://git.trustedfirmware.org/trusted-firmware-m.git/commit/?id=6c5be4a98e4d7055ee49076ca4e515fb4b172e66).

Signed-off-by: Devaraj Ranganna <devaraj.ranganna@arm.com>

urutva committed:
Bump `cryptography` python module version

The requirements.txt mandates 2.4.x version of cryptography python
module which has an issue as it checks if the payload is an instance
of bytes which is actually a bytearray. This issue is fixed in 2.5
version, hence bump up minimum version to 2.5.

Signed-off-by: Devaraj Ranganna <devaraj.ranganna@arm.com>

mark-edgeworth commented on requirements.txt:
```diff
@@ -26,5 +26,5 @@ cmsis-pack-manager>=0.2.3,<0.3.0
 wmi==1.4.9;platform_system=='Windows'
 psutil==5.6.2
-cryptography>=2.4.x,<2.5
+cryptography>=2.5,<=2.8
```
Is there a reason why the upper bound is limited to 2.8? Can we not use '<3'

urutva replied:
Fixed.

mark-edgeworth commented on tools/psa/tfm/bin_utils/imgtool_lib/keys.py:
```diff
@@ -18,42 +18,64 @@
+RSA_KEY_SIZES = [2048, 3072]
+
+# Public exponent
+PUBLIC_EXPONENT = 65537
```
Query: should this be 65536?

urutva replied:
@mark-edgeworth The recommended value for RSA public exponent is 65537.

mark-edgeworth replied:
ok, just checking

mark-edgeworth reviewed (commented):
Some comments in line.
